src/main/scala/is/hail/expr/Relational.scala

```diff
@@ -2441,30 +2439,28 @@ case class TableAggregateByKey(child: TableIR, expr: IR) extends TableIR {
           
           rowKey.setSelect(rowType, keyIndices, current)
 
-          if (rvAggs.nonEmpty) {
-            rvAggs.foreach(_.clear())
+          rvAggs.foreach(_.clear())
+
+          val region = current.region
+          rvb.set(region)
+          rvb.start(globalsType)
+          rvb.addRegionValue(globalsType, partRegion, partGlobalsOff)
+          val globals = rvb.end()
 
```

Comment:
oh weird, these should have been removed. I must have missed them in the rebase. I'll get them in another PR.